scipy/__config__.py.in

```diff
@@ -0,0 +1,139 @@
+# This file is generated by SciPy's build process
+# It contains system_info results at the time of building this package.
+from enum import Enum
+
+__all__ = ["show"]
+_built_with_meson = True
+
+
+class DisplayModes(Enum):
+    stdout = "stdout"
+    dicts = "dicts"
+
+
+def _cleanup(d):
+    """
+    Removes empty values in a `dict` recursively
+    This ensures we remove values that Meson could not provide to CONFIG
+    """
+    if type(d) is dict:
+        return dict(
+            (k, _cleanup(v)) for k, v in d.items() if v and _cleanup(v)
+        )
```

Comment:
I think think this would be easier to read with less brackets
```suggestion
        return {k: _cleanup(v) for k, v in d.items() if v and _cleanup(v)}
```

Reply:
Thanks, refined in [cca74a2](https://github.com/scipy/scipy/pull/17862/commits/cca74a23f015ef08ef424c54d53fd02549892d49)